cmd/api/http/handlers.go

```diff
@@ -0,0 +1,325 @@
+		return
+	}
+
+	responseJSON(w, http.StatusOK, archivedBook)
+}
+
+/* Validates the entry, then stores the entry as a new book. */
+func (h *BookHandler) createBook(w http.ResponseWriter, r *http.Request) {
+	var bookEntry book.Book
+	err := json.NewDecoder(r.Body).Decode(&bookEntry) //Read the Json body and save the entry to bookEntry
+	if err != nil {
+		log.Println(err)
+		errR := book.ErrResponse{
+			Code:    book.ErrResponseBookEntryInvalidJSON.Code,
+			Message: book.ErrResponseBookEntryInvalidJSON.Message + err.Error(),
+		}
+		responseJSON(w, http.StatusBadRequest, errR)
+		return
+	}
+
+	err = book.FilledFields(bookEntry) //Verify if all entry fields are filled.
+	if err != nil {
+		responseJSON(w, http.StatusBadRequest, err)
+		return
+	}
+
+	storedBook, err := h.bookService.CreateBook(bookEntry) //Store the book in the database
+	if err != nil {
+		log.Println(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	responseJSON(w, http.StatusCreated, storedBook)
+}
+
+/* Returns the book with that specific ID. */
+func (h *BookHandler) getBookById(w http.ResponseWriter, r *http.Request) {
+	id, err := isolateId(w, r)
+	if err != nil {
+		return
+	}
+	//Searching for that ID on Book Service:
+	returnedBook, err := h.bookService.GetBook(id)
+	if err != nil {
+		if errors.Is(err, book.ErrResponseBookNotFound) {
+			log.Println(err)
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		log.Println(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	responseJSON(w, http.StatusOK, returnedBook)
+}
+
+/* Returns a list of the stored books. */
+func (h *BookHandler) listBooks(w http.ResponseWriter, r *http.Request) {
+	query := r.URL.Query()
+
+	name := query.Get("name")
+
+	var minPrice32 float32
+	minPriceStr := query.Get("min_price")
+	if minPriceStr != "" {
+		minPrice64, err := strconv.ParseFloat(minPriceStr, 32)
+		if err != nil {
+			responseJSON(w, http.StatusBadRequest, book.ErrResponseQueryPriceInvalidFormat)
+			return
+		}
+		minPrice32 = float32(minPrice64)
+	} else {
+		minPrice32 = 0
+	}
+
+	var maxPrice32 float32
+	maxPriceStr := query.Get("max_price")
+	if maxPriceStr != "" {
+		maxPrice64, err := strconv.ParseFloat(maxPriceStr, 32)
+		if err != nil {
+			responseJSON(w, http.StatusBadRequest, book.ErrResponseQueryPriceInvalidFormat)
+			return
+		}
+		maxPrice32 = float32(maxPrice64)
+	} else {
+		maxPrice32 = book.PriceMax
+	}
+
+	sortBy, sortDirection, valid := extractOrderParams(query)
+	if !valid {
+		responseJSON(w, http.StatusBadRequest, book.ErrResponseQuerySortByInvalid)
+		return
+	}
+
+	archived := false
+	archivedStr := query.Get("archived")
+	if archivedStr == "true" {
+		archived = true
+	}
+
+	page, pageSize, valid := extractPageParams(query)
+	if !valid {
+		responseJSON(w, http.StatusBadRequest, book.ErrResponseQueryPageInvalid)
+		return
+	}
+
+	pagedBooks, err := h.bookService.ListBooks(name, minPrice32, maxPrice32, sortBy, sortDirection, archived, page, pageSize)
```

Comment:
Define a request type as we discussed.
It should be defined inside the `book` package, that will avoid the cyclic dependency issue you had.